common/attest_plugin.c

```diff
@@ -197,6 +198,86 @@ oe_result_t oe_unregister_verifier(oe_verifier_t* plugin)
     return _unregister_plugin(&verifiers, (oe_attestation_role_t*)plugin);
 }
 
+static oe_result_t _get_registered_plugin_format_ids(
+    struct plugin_list_node_t* list,
+    oe_uuid_t** format_ids,
+    size_t* format_ids_length)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    struct plugin_list_node_t* node = NULL;
+    size_t number_of_nodes = 0;
+    oe_uuid_t* ids = NULL;
+
+    if (!format_ids || !format_ids_length)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    *format_ids = NULL;
+    *format_ids_length = 0;
+
+    if (!list)
+        OE_RAISE(OE_NOT_FOUND);
+
+    // Calculate number of nodes
+    node = list;
+    while (node != NULL)
+    {
+        number_of_nodes++;
+        node = node->next;
+    }
+
+    // Allocate memory
+    ids = (oe_uuid_t*)oe_calloc(number_of_nodes, sizeof(oe_uuid_t));
```

Comment:
Consider returning an array of const oe_uuid_t*, this would save some memory.  We may want to refactor how we store the plug-ins in the future.

Reply:
Returning an internal array of const oe_uuid_t* may lead to UAF vulnerability if the app keeps the returned UUID* around after unregistering.

---
In reply to: [365371616](https://github.com/openenclave/openenclave/pull/2402#discussion_r365371616) [](ancestors = 365371616)

Reply:
Does this apply even inside the enclave?

Reply:
Discussed offline.